### PR TITLE
Add HTML5 2D overlay

### PR DESCRIPTION
**Pull request by jason-d-walter**
###### Description:
Enhance the sample project to demonstrate how to implement 2d overlay.

###### Additional Information:
Added new API bindings on the v8 side: 
 
- Gui.rect
- Gui.update_rect
- Gui.destroy_rect

- Gui.bitmap
- Gui.update_bitmap
- Gui.destroy_bitmap

- Vector2
- Vector3

### CODE REVIEW

File: engine/html5_web_api.cpp [hunk 1]
```diff
@@ -41,6 +41,55 @@ void bind_api_web_view(CefRefPtr<CefV8Value> stingray_ns)
 		browser::pick(mouse_pos, from, ray);
 		return CefV8Value::CreateBool(true);
 	});
+
+	bind_api(ns, "create", [](const CefV8ValueList& args)
+    {
```
Comment (jschmidt42) [inline engine/html5_web_api.cpp]: A few whitespace issues. We use tabs.

Comment (jason-d-walter) [reply engine/html5_web_api.cpp]: aggh... ok. I thought I set my editor to tabs but I'll double check.  I have a couple projects where I need to switch back and forth and maybe i didn't catch it here.

File: engine/html5_web_api.cpp [hunk 2]
```diff
@@ -41,6 +41,55 @@ void bind_api_web_view(CefRefPtr<CefV8Value> stingray_ns)
 		browser::pick(mouse_pos, from, ray);
 		return CefV8Value::CreateBool(true);
 	});
+
+	bind_api(ns, "create", [](const CefV8ValueList& args)
```
Comment (jschmidt42) [inline engine/html5_web_api.cpp:45]: I think these APIs where already defined here:

https://github.com/jschmidt42/stingray-html5/blob/603ac28b782e3961fd76ee1da7ec80d1ec300fd6/engine/html5_api_lua.cpp#L223

Comment (jschmidt42) [reply engine/html5_web_api.cpp:45]: Ok that is the JS api. :+1:

**Review by jschmidt42 — APPROVED** [review]
Looks good.

Make sure you use tabs instead of spaces.

I haven't tested it yet.

**Pull request merged**